modules/openapi-generator/src/main/resources/kotlin-spring/dataClass.mustache

```diff
@@ -20,7 +20,7 @@
         {{>interfaceOptVar}}
     {{/optionalVars}}
 {{/discriminator}}
-{{#hasEnums}}{{#vars}}{{#isEnum}}
+{{#hasEnums}}{{#vars}}{{#isEnum}}{{#isPrimitiveType}}
```

Comment:
Interesting. I started looking at this a bit after our Slack conversation, but hadn't thought about how `isPrimitiveType` might fit into things.

One issue with considering enums as primitive is that `$ref` doesn't mean "This is a class/model", it means "I want to reuse this bit of schema in more than one place". So, if an enum which is inline is a primitive type, an enum that is `$ref` should also be a primitive type.

I'll have to think about this a little more and maybe discuss with the core team.

Reply:
Yes, the kotlin test allowed me to notice that an inline enum is basically this in core:
*isEnum = True
*isPrimitive = True
*isString = True

But when you have a referenced enum (with my change), you get:
*isEnum = True

and that's it.

So relying on `isPrimitive` would be the only way for generators to distinguish between the two cases (as far as I've noticed).
It does seem a bit brittle though. I guess `isString` might be ok, but I'm unsure it would cover all cases.